Engine/DataStream.cs

```diff
@@ -52,14 +52,13 @@ public DataStream(IDataFeed feed)
         {
             do
             {
-                TimeSlice timeSlice;
-                while (_feed.Bridge.TryDequeue(out timeSlice))
+                foreach (var timeSlice in _feed.Bridge.GetConsumingEnumerable())
                 {
                     AlgorithmTime = timeSlice.Time;
                     yield return timeSlice.Data;
                 }
-            }
-            while (!_feed.LoadingComplete && !_feed.Bridge.IsEmpty);
```

Comment:
The feed calls Bridge.CompleteAdding(); when finished.
The DataStream checks if finished with Bridge.IsCompleted.
So the bool LoadingComplete is not needed anymore.